### PR TITLE
refactor: split off semantic2 code into semantic2.d

### PR DESCRIPTION
**Pull request by WalterBright**
Just like https://github.com/dlang/dmd/pull/7544

### CODE REVIEW

File: src/win32.mak [hunk 1]
```diff
@@ -162,7 +162,7 @@ FRONT_SRCS=$D/access.d $D/aggregate.d $D/aliasthis.d $D/apply.d $D/argtypes.d $D
 	$D/safe.d $D/blockexit.d $D/permissivevisitor.d $D/transitivevisitor.d $D/parsetimevisitor.d $D/printast.d $D/typesem.d \
 	$D/traits.d $D/utils.d $D/visitor.d $D/libomf.d $D/scanomf.d $D/templateparamsem.d $D/typinf.d \
 	$D/libmscoff.d $D/scanmscoff.d $D/statement_rewrite_walker.d $D/statementsem.d $D/staticcond.d \
-	$D/semantic3.d
+	$D/semantic2 $D/semantic3.d
```
Comment (JinShil) [inline src/win32.mak]: Needs a `.d` file extension.

**Review by JinShil — CHANGES_REQUESTED** [review]

File: src/vcbuild/dmd.visualdproj [hunk 2]
```diff
@@ -543,6 +543,7 @@
    <File path="..\dmd\scanmscoff.d" />
    <File path="..\dmd\scanomf.d" />
    <File path="..\dmd\semantic.d" />
+   <File path="..\dmd\semantic2.d" />
    <File path="..\dmd\sideeffect.d" />
```
Comment (JinShil) [inline src/vcbuild/dmd.visualdproj]: Did we forget to add `..\dmd\semantic3.d` in #7544?

Comment (rainers) [reply src/vcbuild/dmd.visualdproj]: Yes. https://github.com/dlang/dmd/pull/7547

**Review by JinShil — APPROVED** [review]

File: src/dmd/dsymbolsem.d [hunk 3]
```diff
@@ -6,17 +6,12 @@
  * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
  * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
  * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/dsymbolsem.d, _dsymbolsem.d)
+ * Documentation:  https://dlang.org/phobos/dmd_dsymbolsem.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/dsymbolsem.d
```
Comment (wilzbach) [inline src/dmd/dsymbolsem.d]: https://dlang.org/phobos-prerelease/dmd_semantic2.html

Ddoc's auto-escaping "wins" again:

image

**Pull request merged**